Engine/controller.py

```diff
@@ -5,111 +5,103 @@
+        self.timestamp = -1
+        self.robots = [Robot(robot_id) for robot_id in range(config['ENGINE']['max_robot_id'])]
+        for robot in self.robots: robot.velocity_regulator = VelocityRegulator()
 
-        for robot in self.robots:
-            robot.position_regulator = PositionRegulator()
-            robot.velocity_regulator = VelocityRegulator()
-
-    def update(self, track_frame: Dict, engine_cmds: List[EngineCommand]):
-        self.timestamp = track_frame['timestamp']
-        our_team_color = str(TeamColorService().our_team_color)
 
-        for robot in self:
-            robot.is_active = False
+    def update(self, track_frame: Dict[str, Any], engine_cmds: List[EngineCommand]):
 
-        for robot in track_frame[our_team_color]:
+        self.timestamp = track_frame['timestamp']
 
-            self[robot['id']].is_active = True
+        for robot in track_frame[config['GAME']['our_color']]:
             self[robot['id']].pose = robot['pose']
             self[robot['id']].velocity = robot['velocity']
 
+        for robot in self.robots:
+            robot.engine_cmd = None
+
         for cmd in engine_cmds:
             self[cmd.robot_id].engine_cmd = cmd
-            if self[cmd.robot_id].raw_path is None:
-                self[cmd.robot_id].path = None
 
     def execute(self) -> RobotState:
         commands = {}
-        active_robots = [robot for robot in self if robot.is_active]
-
-        for robot in active_robots:
-            if robot.raw_path is None:
-                continue
-            robot.path, robot.target_speed = path_smoother(robot)
-
-            if robot.position_error.norm < 100 and robot.target_speed == 0:
-                cmd = robot.position_regulator.execute(robot)
-                robot.velocity_regulator.reset()
-            else:
-                cmd = robot.velocity_regulator.execute(robot)
-
-            self.ui_send_queue.put_nowait(DebugCommandFactory.plot_point("mm/s",
-                                                                         "robot {} cmd speed".format(robot.robot_id),
-                                                                         [time.time()],
-                                                                         [cmd.norm]))
-            self.ui_send_queue.put_nowait(DebugCommandFactory.plot_point("mm/s",
-                                                                         "robot {} kallman speed".format(robot.robot_id),
-                                                                         [time.time()],
-                                                                         [robot.velocity.norm]))
-
-            commands[robot.robot_id] = self._put_in_robots_referential(robot, cmd)
+
+        for robot in self.active_robots:
+            robot.path, robot.target_speed = path_smoother(robot.raw_path, robot.cruise_speed, robot.end_speed)
+            commands[robot.id] = robot.velocity_regulator.execute(robot)
```

Comment:
I remove the position regulator according to a comment from @wonwon0 . Could be readded if needed.